test/e2e/plugins_compatibility/plugins_compatibility_helper_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// plugincompatibility provides plugins compatibility E2E test cases
+package plugincompatibility
+
+import (
+	"os"
+	"strings"
+
+	"github.com/aunum/log"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-cli/test/e2e/framework"
+)
+
+// PluginsForCompatibilityTesting search for test-plugin-'s from the test central repository and returns all test-plugin-'s
+func PluginsForCompatibilityTesting(tf *framework.Framework) []string {
+	list, err := tf.PluginCmd.SearchPlugins("")
+	Expect(err).To(BeNil(), "should not occur any error while searching for plugins")
+	testPlugins := make([]string, 0)
+	for _, plugin := range list {
+		if strings.HasPrefix(plugin.Name, framework.TestPluginsPrefix) {
+			testPlugins = append(testPlugins, plugin.Name)
+		}
+	}
+	if len(testPlugins) == 0 {
+		url := os.Getenv(framework.TanzuCliE2ETestCentralRepositoryURL)
+		log.Errorf("there are no test-plugin-'s in test central repo:%s make sure, its valid test central repo with test-plugins", url)
```

Comment:
E2e tests should fail if this is the case instead of just logging an error.

Reply:
Test plugins might not be published yet, right?

Reply:
If it's not published, we still want to catch that scenario and fail the pipeline. If the user is running these tests means, the user is expected to have plugins published already.

Also, this might give some false positive when by mistake someone removes all plugins from the test repository and pipeline always succeeds without anyone noticing it from logs.

Reply:
Got it, I will make it fail if no test plugins available.